lighthouse-core/config/config.js

```diff
@@ -182,11 +182,9 @@ function cleanFlagsForSettings(flags = {}) {
   const settings = {};
 
   for (const key of Object.keys(flags)) {
-    // @ts-ignore - intentionally testing some keys not on defaultSettings to discard them.
-    if (typeof constants.defaultSettings[key] !== 'undefined') {
```

Comment:
> I seem to remember being very careful about in vs undefined for constants.defaultSettings but that might have just been on the delete vs. = undefined side.

yeah, I was thinking since it's just the constants this should be ok (we won't have set anything to `undefined` to try to mark them as unset), but I can also switch it to an undefined check. It doesn't help the compiler either way since we have to do `@ts-ignore` on the next line regardless :)